test/tests/functional/pbs_qsub_remove_files.py

```diff
@@ -54,6 +54,123 @@ def test_remove_file_when_job_succeeds(self):
         j.set_sleep_time(1)
         sub_dir = DshUtils().mkdtemp(uid=TEST_USER.uid)
         jid = self.server.submit(j, submit_dir=sub_dir)
+        self.server.expect(JOB, {'Remove_Files': 'oe'}, id=jid)
+        self.server.log_match(jid + ";dequeuing from", max_attempts=10)
+        file_count = len([name for name in os.listdir(
+            sub_dir) if os.path.isfile(os.path.join(sub_dir, name))])
+        self.assertEqual(0, file_count)
+
+    def test_remove_file_sandbox_private(self):
+        """
+        submit a sleep job and make sure that the std_files
+        are getting deleted when remove_files option is used
+        and job is submitted with -Wsandbox=private.
+        """
+        j = Job(TEST_USER)
+        j.set_attributes({ATTR_R: 'oe'})
+        j.set_attributes({ATTR_sandbox: 'private'})
+        j.set_sleep_time(1)
+        sub_dir = DshUtils().mkdtemp(uid=TEST_USER.uid)
+        jid = self.server.submit(j, submit_dir=sub_dir)
+        self.server.expect(JOB, {'Remove_Files': 'oe'}, id=jid)
+        self.server.log_match(jid + ";dequeuing from", max_attempts=10)
+        file_count = len([name for name in os.listdir(
+            sub_dir) if os.path.isfile(os.path.join(sub_dir, name))])
+        self.assertEqual(0, file_count)
+
+    def test_remove_files_output_file(self):
+        """
+        submit a job with -Ro option and make sure the output file
+        gets deleted after job finishes
+        """
+        j = Job(TEST_USER)
+        j.set_attributes({ATTR_R: 'o'})
+        j.set_sleep_time(1)
+        sub_dir = DshUtils().mkdtemp(uid=TEST_USER.uid)
+        jid = self.server.submit(j, submit_dir=sub_dir)
+        self.server.expect(JOB, {'Remove_Files': 'o'}, id=jid)
+        self.server.log_match(jid + ";dequeuing from", max_attempts=10)
+        file_count = len([name for name in os.listdir(
+            sub_dir) if os.path.isfile(os.path.join(sub_dir, name))])
+        self.assertEqual(1, file_count)
+
+    def test_remove_files_error_file(self):
+        """
+        submit a job with -Re option and make sure the error file
+        gets deleted after job finishes and works with direct_write
+        """
+        j = Job(TEST_USER)
+        j.set_attributes({ATTR_k: 'de'})
+        j.set_attributes({ATTR_R: 'e'})
+        j.set_sleep_time(1)
+        sub_dir = DshUtils().mkdtemp(uid=TEST_USER.uid)
+        mapping_dir = DshUtils().mkdtemp(uid=TEST_USER.uid)
+        self.mom.add_config(
+            {'$usecp': self.server.hostname + ':' + sub_dir
+             + ' ' + mapping_dir})
+        self.mom.restart()
+        jid = self.server.submit(j, submit_dir=sub_dir)
+        self.server.expect(JOB, {'Remove_Files': 'e'}, id=jid)
+        self.server.log_match(jid + ";dequeuing from", max_attempts=10)
+        file_count = len([name for name in os.listdir(
+            mapping_dir) if os.path.isfile(os.path.join(mapping_dir, name))])
+        self.assertEqual(1, file_count)
```

Comment:
This test makes sure that only one file exists at the end of the job. But we need to make sure that it is the output file.